src/serac/physics/materials/thermal_functional_material.hpp

```diff
@@ -0,0 +1,278 @@
+      : density_(density), specific_heat_capacity_(specific_heat_capacity), conductivity_(conductivity)
+  {
+    SLIC_ERROR_ROOT_IF(density_ < 0.0, "Density must be positive in the linear conductor material model.");
+
+    SLIC_ERROR_ROOT_IF(specific_heat_capacity_ < 0.0,
+                       "Specific heat capacity must be positive in the linear conductor material model.");
+
+    // Check that the conductivity tensor is symmetric
+    for (int i = 0; i < dim; ++i) {
+      for (int j = i + 1; j < dim; ++j) {
+        SLIC_ERROR_ROOT_IF(std::abs(conductivity_(i, j) - conductivity_(j, i)) > 1.0e-7,
+                           "Conductivity tensor must be symmetric for the linear conductor material model.");
+      }
+    }
+
+    // Check for positive definite conductivity using Sylvester's criterion
+    // The upper left corner sub-matrices must have a positive determinant
+    SLIC_ERROR_ROOT_IF(conductivity_(0, 0) < 0.0,
+                       "Conductivity tensor must be positive definite for the linear conductor material model.");
+
+    SLIC_ERROR_ROOT_IF(det(conductivity_) < 0.0,
+                       "Conductivity tensor must be positive definite for the linear conductor material model.");
+
+    if (dim == 3) {
+      auto subtensor_2D = make_tensor<2, 2>([this](int i, int j) { return conductivity_(i, j); });
+      SLIC_ERROR_ROOT_IF(det(subtensor_2D) < 0.0,
+                         "Conductivity tensor must be positive definite for the linear conductor material model.");
+    }
+  }
+
+  /**
+   * @brief Function defining the thermal flux (constitutive response)
+   *
+   * @tparam T1 type of the temperature (e.g. tensor or dual type)
+   * @tparam T2 type of the temperature gradient (e.g. tensor or dual type)
+   * @param temperature_gradient Gradient of the temperature (du_dx)
+   * @return The thermal flux of the material model
+   */
+  template <typename T1, typename T2>
+  SERAC_HOST_DEVICE auto operator()(T1& /* temperature */, T2& temperature_gradient) const
+  {
+    return -1.0 * conductivity_ * temperature_gradient;
+  }
+
+  /**
+   * @brief The density (mass per volume) of the material model
+   *
+   * @return The density
+   */
+  SERAC_HOST_DEVICE double density(tensor<double, dim>& /* x */) const { return density_; }
+
+  /**
+   * @brief The specific heat capacity (heat capacity per unit mass) of the material model
+   *
+   * @tparam T Type of the temperature variable
+   */
+  template <typename T>
+  SERAC_HOST_DEVICE double specificHeatCapacity(tensor<double, dim>& /* x */, T& /* temperature */) const
+  {
+    return specific_heat_capacity_;
+  }
+
+private:
+  /// Density
+  double density_ = 1.0;
+
+  /// Specific heat capacity
+  double specific_heat_capacity_ = 1.0;
+
+  /// Constant thermal conductivity
+  tensor<double, dim, dim> conductivity_ = {{{1.0, 0.0, 0.0}, {0.0, 1.0, 0.0}, {0.0, 0.0, 1.0}}};
```

Comment:
making these private members disables aggregate initialization, so I don't think that the "default" values specified here do anything.

Reply:
Yeah, this is just me being probably too cautious. I'll remove them.